### PR TITLE
JSAB2-27 Login frontend

### PR DESCRIPTION
**Pull request by Kyya**
+ build login frontend using mocked user
+ set login screen as the default navigation screen

test username: kyya
test password: password

### CODE REVIEW

File: frontend/src/Login/authReducer.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { LOGIN } from './actionCreator';
+
+const defaultState = {
+  isLogin: false,
```
Comment (evasimonyi) [inline frontend/src/Login/authReducer.js]: isLoggedIn? or isLoggingIn? which one is it? please rename, actually I can't decide what you wanted, the process, like isLoading, or the state of the user, who isLoggedIn. I guess it's the logged in status, right?

```suggestion
  isLoggedIn: false,
```

File: frontend/src/Login/authReducer.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { LOGIN } from './actionCreator';
+
+const defaultState = {
```
Comment (evasimonyi) [inline frontend/src/Login/authReducer.js]: please call it initialState

```suggestion
const initialState = {
```

File: frontend/src/Login/index.jsx [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  },
+});
+
+const mockedUser = {
+  username: 'kyya',
+  password: 'password',
+};
+
+function showAlert(text) {
+  Alert.alert('Tips', text);
```
Comment (evasimonyi) [inline frontend/src/Login/index.jsx]: please rename from tips, I'm not sure what's the best practice for this, but I've never seen `Tips` in an application

File: frontend/src/Login/index.jsx [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+}
+
+function Login() {
+  const dispatch = useDispatch();
+  const navigation = useNavigation();
+  const [username, setUsername] = useState('');
+  const [password, setPassword] = useState('');
+
+  function handleSubmit() {
+    const name = username.toLowerCase();
```
Comment (evasimonyi) [inline frontend/src/Login/index.jsx]: why do you want it to be lowerCased?

File: frontend/src/Login/CardTextInput.jsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import { CardView } from '../common/components';
+
+const style = {
+  height: 40,
+  margin: 5,
+  paddingStart: 10,
+  fontSize: 14,
+};
+
+function CardTextInput({
```
Comment (evasimonyi) [inline frontend/src/Login/CardTextInput.jsx]: I'm not sure why do we use so much `Card`-s in the naming :D but you can just call it InputField or something like that. Of course this is fine too:)

File: frontend/src/Login/authReducer.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+  isLogin: false,
+};
+
+export default function auth(state = defaultState, action) {
+  switch (action.type) {
+    case LOGIN:
+      return { ...state, isLogin: true };
```
Comment (evasimonyi) [inline frontend/src/Login/authReducer.js]:
```suggestion
      return { ...state, isLoggedIn: true };
```
I guess you wanted this, right?

File: frontend/src/Login/authReducer.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
+import { LOGIN } from './actionCreator';
+
+const defaultState = {
+  isLogin: false,
+};
+
+export default function auth(state = defaultState, action) {
```
Comment (evasimonyi) [inline frontend/src/Login/authReducer.js]:
```suggestion
export default function auth(state = initialState, action) {
```

File: frontend/src/Login/index.jsx (hunk 4 above)
Comment (Kyya) [reply frontend/src/Login/index.jsx]: When someone types username 'Kyya', he cannot logged in...but he should

File: frontend/src/Login/CardTextInput.jsx (hunk 5 above)
Comment (Kyya) [reply frontend/src/Login/CardTextInput.jsx]: I made a new style for TextInput based on card style,
so I call it CardTextInput, If we have a better name for it, I will use it.

**Pull request merged**